File: game_client/Assets/Scripts/ClientProtocol/Common/RechargeRebateEnum/ERechargeRebateType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.RechargeRebateEnum
{

/// <summary>
/// 充值返利类型枚举
/// </summary>
public enum ERechargeRebateType {
	[InspectorName("DAILY - idx[0] - 每日充值")]
	DAILY, //0 ==== 每日充值
	[InspectorName("TOTAL - idx[1] - 累计充值")]
	TOTAL, //1 ==== 累计充值
	[InspectorName("DAYS - idx[2] - 累天充值")]
	DAYS, //2 ==== 累天充值
}

public class ERechargeRebateTypeComparer : IEqualityComparer<ERechargeRebateType>{
	public bool Equals(ERechargeRebateType x, ERechargeRebateType y) { return x == y; }
	public int GetHashCode(ERechargeRebateType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 3;
}
}

File: game_server/ServerProtocol/src/Common/RechargeRebateEnum/ERechargeRebateType.java
package Common.RechargeRebateEnum;

/*********
 * 充值返利类型枚举
 **/
public enum ERechargeRebateType {
	DAILY, //0 ==== 每日充值
	TOTAL, //1 ==== 累计充值
	DAYS, //2 ==== 累天充值
	;
public static final ERechargeRebateType[]  ERechargeRebateType_Values = ERechargeRebateType.values();
public static final int ERechargeRebateType_Length = ERechargeRebateType_Values.length;
public static ERechargeRebateType ERechargeRebateType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= ERechargeRebateType_Length){ return null; }
	return ERechargeRebateType_Values[_ivalue];
}
}

